azure/services/virtualmachineimages/images.go

```diff
@@ -180,14 +180,20 @@ func (s *Service) getSKUAndVersion(ctx context.Context, location, publisher, off
 
 // getUbuntuOSVersion returns the default Ubuntu OS version for the given Kubernetes version.
 func getUbuntuOSVersion(major, minor, patch uint64) string {
-	// Default to Ubuntu 20.04 LTS, except for k8s versions which have only 18.04 reference images.
-	osVersion := "2004"
+	// Default to Ubuntu 22.04 LTS, except for k8s versions which have only 20.04 or 18.04 reference images.
+	osVersion := "2204"
 	if (major == 1 && minor == 21 && patch < 2) ||
 		(major == 1 && minor == 20 && patch < 8) ||
 		(major == 1 && minor == 19 && patch < 12) ||
 		(major == 1 && minor == 18 && patch < 20) ||
 		(major == 1 && minor < 18) {
 		osVersion = "1804"
+	} else if (major == 1 && minor == 25 && patch < 3) ||
+		(major == 1 && minor == 24 && patch < 7) ||
+		(major == 1 && minor == 23 && patch < 13) ||
```

Comment:
My understanding is that cgroupv2 is enabled by default on ubuntu 22.04. If so, I don't think we should publish images for < k8s 1.24.

Reply:
Right you are (according to my Googling), thanks for pointing this out. On the [Kubernetes cgroups doc](https://kubernetes.io/docs/concepts/architecture/cgroups/#cgroup-v2), it says:

> **FEATURE STATE**: Kubernetes v1.25 [stable]

Should we restrict Ubuntu 22.04 to >= v1.25 then?